applications/ConstitutiveLawsApplication/custom_utilities/advanced_constitutive_law_utilities.h

```diff
@@ -478,6 +479,86 @@ class KRATOS_API(CONSTITUTIVE_LAWS_APPLICATION) AdvancedConstitutiveLawUtilities
         BoundedMatrixVoigtType &rNewOperator
     );
 
+    /**
+     * @brief This computes a material property according to a certain
+     * nodal TEMPERATURE table
+     */
+    static double GetValueFromTable(
+        const Variable<double>& rIndependentVariable,
+        const Variable<double>& rDependentVariable,
+        ConstitutiveLaw::Parameters& rParameters
+        )
+    {
+        // Get material properties from constitutive law parameters
+        const Properties& r_properties = rParameters.GetMaterialProperties();
+
+        // Get geometry and Gauss points data
+        const auto& r_geometry = rParameters.GetElementGeometry();
+        const auto& r_N        = rParameters.GetShapeFunctionsValues();
+
+        // Compute the independent variable at the Gauss point
+        double independent_at_gauss = 0.0;
+        for (std::size_t i = 0; i < r_N.size(); ++i) {
+            const double val = r_geometry[i].FastGetSolutionStepValue(rIndependentVariable);
+            independent_at_gauss += val * r_N[i];
+        }
+
+        // Retrieve the dependent variable from the table
+        const auto& r_table = r_properties.GetTable(rIndependentVariable, rDependentVariable);
+        return r_table.GetValue(independent_at_gauss);
+    }
+
+    /**
+     * @brief This retrieves an interpolated nodal variable to a GP
+     */
+    static double CalculateInGaussPoint(
+        const Variable<double>& rVariableInput,
+        ConstitutiveLaw::Parameters& rParameters,
+        unsigned int step = 0
+        )
+    {
+        return CalculateInGaussPoint(rVariableInput, rParameters.GetElementGeometry(), rParameters.GetShapeFunctionsValues());
+    }
+
+    /**
+     * @brief This retrieves an interpolated nodal variable to a GP
+     */
+    static double CalculateInGaussPoint(
+        const Variable<double>& rVariableInput,
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunctionsValues,
+        unsigned int step = 0
+        )
+    {
+
+        const std::size_t number_of_nodes = rGeometry.size();
+        double result = 0;
+
+        for (IndexType i = 0; i < number_of_nodes; ++i) {
+            result += rShapeFunctionsValues[i] * rGeometry[i].FastGetSolutionStepValue(rVariableInput, step);
+        }
+        return result;
+    }
+
+    /**
+     * @brief This substracts the thermal strain contribution to a vector
+     */
+    static void SubstractThermalStrain(
+        ConstitutiveLaw::StrainVectorType& rStrainVector,
+        const double ReferenceTemperature,
+        ConstitutiveLaw::Parameters& rParameters
+        )
+    {
+        double alpha = rParameters.GetMaterialProperties()[THERMAL_EXPANSION_COEFFICIENT];
```

Comment:
In our experience in casting, this one is also temperature dependent...

Reply:
Interesting, in order to not increase the already big PR, I'll add this in the future if you do not mind :D

Reply:
Considering that `Properties` support tables, this shouldn't be a problem right?

Reply:
not at all... I can do it in this PR if you think that it is important

Reply:
this needs the same solution with the virtual fct no?

Reply:
Maybe we can leave it for a  future PR but I agree @ddiezrod that it should be possible to make these kind of material properties temperature dependent. Hence we'll need tables for sure.

Reply:
I'll do it this afternoon or tomorrow

Reply:
done :D